cmd/bucket-replication.go

```diff
@@ -407,83 +452,132 @@ func getReplicationAction(oi1 ObjectInfo, oi2 minio.ObjectInfo) replicationActio
 		oi1.VersionID != oi2.VersionID ||
 		oi1.Size != oi2.Size ||
 		oi1.DeleteMarker != oi2.IsDeleteMarker ||
-		!oi1.ModTime.Equal(oi2.LastModified) {
+		oi1.ModTime.Unix() != oi2.LastModified.Unix() {
 		return replicateAll
 	}
+
 	if oi1.ContentType != oi2.ContentType {
 		return replicateMetadata
 	}
+
 	if oi1.ContentEncoding != "" {
 		enc, ok := oi2.Metadata[xhttp.ContentEncoding]
 		if !ok || strings.Join(enc, "") != oi1.ContentEncoding {
 			return replicateMetadata
 		}
 	}
+
+	// Do not compare oi2.UserMetadata - it does not contain `x-amz-meta-` prefix.
+
 	// compare metadata on both maps to see if meta is identical
 	for k1, v1 := range oi1.UserDefined {
-		if v2, ok := oi2.UserMetadata[k1]; ok && v1 == v2 {
+		if v2, ok := oi2.Metadata[strings.ToLower(k1)]; ok && v1 == strings.Join(v2, "") {
 			continue
 		}
 		if v2, ok := oi2.Metadata[k1]; ok && v1 == strings.Join(v2, "") {
 			continue
 		}
 		return replicateMetadata
 	}
-	for k1, v1 := range oi2.UserMetadata {
-		if v2, ok := oi1.UserDefined[k1]; !ok || v1 != v2 {
-			return replicateMetadata
-		}
-	}
+
 	for k1, v1slc := range oi2.Metadata {
 		v1 := strings.Join(v1slc, "")
-		if k1 == xhttp.ContentEncoding { //already compared
+		if k1 == xhttp.ContentEncoding { // already compared
 			continue
 		}
-		if v2, ok := oi1.UserDefined[k1]; !ok || v1 != v2 {
+		if k1 == xhttp.ContentType { // already compared
+			continue
+		}
+		v2, ok := oi1.UserDefined[k1]
+		if !ok {
+			v2, ok = oi1.UserDefined[strings.ToLower(k1)]
+			if !ok {
+				return replicateMetadata
+			}
+		}
```

Comment:
since X-Amz-Replication-Status would not be filtered as per https://github.com/minio/minio-go/blob/546cb78a3d221a947d877b7015f62e2e53fb033e/utils.go#L183 that needs to be removed from comparison too